test/ranges.jl

```diff
@@ -305,6 +305,7 @@ end
         @test findfirst(==(7), 1:2:10) == 4
         @test findfirst(==(10), 1:2:10) == nothing
         @test findfirst(==(11), 1:2:10) == nothing
+        @test findfirst(==(-7), 1:-1:-10) == 9
```

Comment:
maybe add a test where the range is empty, now that this is specifically addressed in the code?